### PR TITLE
Fix rustdoc generation of Windows and Mac on docs.rs, fix #503

### LINKED ISSUE #503: Doc.rs of macOS and Windows aren't built
The docs of Mac and Windows are missing after 0.13.
This is probably because of `thiserror`. Its recent patch version only generates linux target.
And on Windows, it's also because of sys-info. Since it's only used for detecting windows 7, we could add our own helper function instead.

### PR DESCRIPTION
**Pull request by wusyong**
### What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [x] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [x] No

### Checklist
- [x] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [x] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [x] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -78,5 +78,5 @@ sys-info = "0.9"
 [target."cfg(any(target_os = \"ios\", target_os = \"macos\"))".dependencies]
 cocoa = "0.24"
 core-graphics = "0.22"
-objc = { version = "0.2", features = [ "exception" ] }
+objc = "0.2"
```
Comment (lucasfernog) [inline Cargo.toml:82]: I really think you should keep the `exception` as a default feature flag, disabled on docs.rs, this was part of the audit

**Comment by lucasfernog** [comment]
Are you sure it builds without the default features? I thought we only needed to use the objc/exception feature as a default feature.

**Comment by wusyong** [comment]
No, it's not. That's why I set the features need to use in docs.rs
That being said, I update the scope of default features like you said.

File: Cargo.toml [hunk 2]
```diff
@@ -78,5 +78,5 @@ sys-info = "0.9"
 [target."cfg(any(target_os = \"ios\", target_os = \"macos\"))".dependencies]
 cocoa = "0.24"
 core-graphics = "0.22"
-objc = { version = "0.2", features = [ "exception" ] }
+objc = { version = "0.2", features = ["exception"] }
```
Comment (lucasfernog) [inline Cargo.toml]: you still need to remove the feature here right? since it's enabled with default features now

**Review by lucasfernog — APPROVED** [review]

**Pull request merged**